go/vt/tabletmanager/reparent.go

```
@@ -18,14 +23,191 @@ import (
+
+	// Execute state change to master by force-updating only the local copy of the
+	// tablet record. The actual record in topo will be updated later.
+	log.Infof("TabletExternallyReparented: executing change callback for state change to MASTER")
+	oldTablet := *tablet.Tablet
+	newTablet := oldTablet
+	newTablet.Type = topo.TYPE_MASTER
+	newTablet.State = topo.STATE_READ_WRITE
+	newTablet.Health = nil
+	agent.setTablet(topo.NewTabletInfo(&newTablet, -1))
+	if err := agent.updateState(ctx, &oldTablet, "TabletExternallyReparented"); err != nil {
+		return fmt.Errorf("TabletExternallyReparented: failed to change tablet state to MASTER: %v", err)
+	}
+
+	// Directly write the new master endpoint in the serving graph.
+	// We will do a true rebuild in the background soon, but in the meantime,
+	// this will be enough for clients to re-resolve the new master.
+	event.DispatchUpdate(ev, "writing new master endpoint")
+	log.Infof("TabletExternallyReparented: writing new master endpoint to serving graph")
+	ep, err := tablet.EndPoint()
+	if err != nil {
+		return fmt.Errorf("failed to generate EndPoint for tablet %v: %v", tablet.Alias, err)
+	}
+	err = topo.UpdateEndPoints(ctx, agent.TopoServer, tablet.Alias.Cell,
+		si.Keyspace(), si.ShardName(), topo.TYPE_MASTER,
+		&topo.EndPoints{Entries: []topo.EndPoint{*ep}})
+	if err != nil {
+		return fmt.Errorf("TabletExternallyReparented: failed to update master endpoint: %v", err)
+	}
+
+	// Start the finalize stage with a background context, but connect the trace.
+	go agent.finalizeTabletExternallyReparented(
+		trace.CopySpan(agent.batchCtx, ctx), si, ev)
+
+	return nil
+}
+
+// finalizeTabletExternallyReparented performs slow, synchronized reconciliation
+// tasks that ensure topology is self-consistent, and then marks the reparent as
+// finished by updating the global shard record.
+func (agent *ActionAgent) finalizeTabletExternallyReparented(ctx context.Context, si *topo.ShardInfo, ev *events.Reparent) (err error) {
```

this returns an error but is called n the background so the error is ignored?